### PR TITLE
TW-777 Push notifications

### PR DESCRIPTION
**Pull request by lendihop**
https://madfish.atlassian.net/jira/software/projects/TW/boards/4?selectedIssue=TW-777

### CODE REVIEW

File: src/hooks/use-push-notifications.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+import { useEffect } from 'react';
+import { PermissionsAndroid } from 'react-native';
+import PushNotification from 'react-native-push-notification';
+
+import { isAndroid } from 'src/config/system';
+import { isDefined } from 'src/utils/is-defined';
+
+export const usePushNotifications = () => {
+  useEffect(() => {
+    requestUserPermission();
```
[the remark this answers is not in the document]
Comment (lendihop) [reply src/hooks/use-push-notifications.ts:13]: actually there was a little problem with that on android.. made refactor

**Review by keshan3262 — APPROVED** [review]

File: index.js [hunk 2]
```diff
@@ -3,9 +3,16 @@ import 'react-native-gesture-handler';
...
 
 import { name as appName } from './app.json';
 import { App } from './src/app/app';
+import { isAndroid } from './src/config/system';
+
+(async () => {
+  if (isAndroid) {
+    await PermissionsAndroid.request(PermissionsAndroid.PERMISSIONS.POST_NOTIFICATIONS);
+  }
+})();
```
[the remark this answers is not in the document]
Comment (lendihop) [reply index.js:16]: its ok, pls check temple-requirements chat

**Review by keshan3262 — APPROVED** [review]

File: src/hooks/use-push-notifications.ts [hunk 3]
```diff
@@ -1,27 +1,47 @@
...
+  const { trackEvent } = useAnalytics();
+
   useEffect(() => {
-    requestUserPermission();
+    requestUserPermission(getFcmToken);
     const unsubscribe = handleForegroundNotifications();
 
     return unsubscribe;
-  });
+  }, []);
```
[the remark this answers is not in the document]
Comment (lendihop) [reply src/hooks/use-push-notifications.ts:22]: this fn may change only when user disables analytics in the settings as i see

**Review by NovakVlad2424 — APPROVED** [review]
ios done: 1.15_806

**Review by IrynaKhyzhynska — APPROVED** [review]
android done 1.15_806

**Review by vladvn1905 — APPROVED** [review]
ios/android done build: 1.15_806

**Pull request merged**